### ISSUE TITLE
Realms.RequiredAttribute is broken

### ISSUE THREAD

**Issue by kraeg75** [realm/realm-dotnet#2085]
The second assert fails.  The expected value is **2**, but Realms is producing **0**.

It works fine in version 4.3.0 and is broken in version 5.0.1 onwards.

```csharp
using System;
using NUnit.Framework;
using Realms;

namespace RealmTests
{
	public class RealmTests
	{
		public class Item : RealmObject
		{
			public int? UnrequiredNumber { get; set; }

			[Required]
			public int? RequiredNumber { get; set; }
		}

		[Test]
		public void RequiredTest()
		{
			var realm = Realm.GetInstance(new InMemoryConfiguration(Guid.NewGuid().ToString()));

			var item = new Item { UnrequiredNumber = 1, RequiredNumber = 2 };
			realm.Write(() => realm.Add(item));

			Assert.AreEqual(1, item.UnrequiredNumber);
			Assert.AreEqual(2, item.RequiredNumber);
		}
	}
}

**Comment by nirinchev** [comment]
I was able to repro an issue related to applying `[Required]` on nullable integral types and we'll look into a fix. In the meantime, you should be able to avoid it by using the non-nullable version of `int`:

```csharp
public class Item : RealmObject
{
    public int? UnrequiredNumber { get; set; }

    public int RequiredNumber { get; set; }
}
```

**Comment by papafe** [comment]
I could not reproduce the error on the latest available Realm version (10.1.3), so I am closing the ticket. Please feel free to reopen the ticket if this is still an issue for you.
